src/ResourceManager/Resources/Commands.Resources/RoleAssignments/ValidateScope.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.Common
+{
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+    public class ValidateScope : ValidateArgumentsAttribute
+    {
+        protected override void Validate(object arguments, EngineIntrinsics engineIntrinsics)
+        {
+            if (arguments != null)
```

Comment:
I tried to do so in first iteration but later i realized that for "providers" section, we could have more than 1 corresponding value.
also that won't help for this case: we have two name parts, but neither has a value, this will cause the number of parts being even but still invalid. like: "/subscriptions/providers/"

so in iteration#2 i made more strict check.

---
In reply to: [118150608](https://github.com/Azure/azure-powershell/pull/4019#discussion_r118150608) [](ancestors = 118150608)